pkg/provisioner/controller.go

```diff
@@ -707,10 +707,14 @@ func (c *obcController) handleUpdateClaim(key string, obc *v1alpha1.ObjectBucket
 }
 
 func overwriteAdditionalConfig(srcAdditionalConfig, destAdditionalConfig map[string]string) {
-	for key, _ := range destAdditionalConfig {
-		delete(destAdditionalConfig, key)
-	}
-	for key, value := range srcAdditionalConfig {
-		destAdditionalConfig[key] = value
+	if destAdditionalConfig != nil {
+		for key, _ := range destAdditionalConfig {
+			delete(destAdditionalConfig, key)
+		}
+		for key, value := range srcAdditionalConfig {
+			destAdditionalConfig[key] = value
+		}
+	} else {
+		destAdditionalConfig = srcAdditionalConfig
```

Comment:
I'm not sure if this is the source of the issue, but the copy must be a deep copy.

```suggestion
	if destAdditionalConfig != nil {
		for key, _ := range destAdditionalConfig {
			delete(destAdditionalConfig, key)
		}
	}
	for key, value := range srcAdditionalConfig {
		destAdditionalConfig[key] = value
	}
```